### ISSUE TITLE
[Bug]: Counting tiles for deciding the most used rail type doesn't take into account owners

### ISSUE THREAD

**Issue by lcd047** [OpenTTD/OpenTTD#9721]
### Version of OpenTTD

12.1, Linux

### Expected result

When "Default rail type" is set to "most used", only tiles owned by the current company should be counted.  Tiles owned by other companies should never affect the outcome.

### Actual result

All rail tiles are counted, regardless of owners.  As a result, if you quit a network game then re-join later, the default rail type when you press "A" depends on what the other people have done in the mean time, which is annoying.  This should never happen, what other people do should be irrelevant for client defaults.

### Steps to reproduce

1. Set "Default rail type" to "most used"
2. Start a network game
3. Create a company, build electrified rails on a single tile, then quit
4. Create another company, build maglev rails on two tiles, then quit
5. Join the first company again, and press "A".

The result is a GUI menu for maglev instead of the expected electrified rails.

**Comment by frosch123** [comment]
Currently "default rail type" is only evaluated once when joining/loading a game. After that the active railtype only changes on explicit player action.

How should this behave, when it was company dependant?
* If you first join as spectator, and then join a company, I guess evaluation of "most used" shall happen when joining the company.
* If you leave a company and rejoin the same or another company: should it keep the last manually choosen railtype, or reevaluate the "most used" type?

**Comment by lcd047** [comment]
Yes, I suppose the default type should be re-evaluated whenever the current company changes.

Whether the last manually chosen type should be used is less obvious.  I mean, ideally it should be used if it exists, then "Default rail type" should be evaluated as a fallback.  But that would involve keeping track of the last used type per company per game, on the client side.  If that's too complicated I suppose "Default rail type" is good enough, since it doesn't need to keep state.

**Comment by lcd047** [comment]
Tangentially related: now that we can have multiple road types perhaps "Default rail type" should also apply to roads, for consistency of the UI.

**Comment by PeterN** [comment]
Fixed by #15489
